### PR TITLE
ci: more improvements

### PR DESCRIPTION
**Pull request by bmwill**

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -61,10 +61,10 @@ jobs:
         with:
           path: ~/.cargo/registry/src/**/librocksdb-sys-*
       - name: cargo test
-        uses: actions-rs/cargo@v1
-        with:
-          command: test
-          args: --all-features
+        run: |
+          cargo build --all-features --lib
+          cargo test --all-features --no-run -j 1
```
Comment (mystenmark) [inline .github/workflows/rust.yml:66]: i would suggest higher than `-j 1` - can we try 8 and see if it runs ok?

Comment (bmwill) [reply .github/workflows/rust.yml:66]: yeah i was just trying to see if i could get it to pass, it didn't. But with the self-hosted runners we'll want this to be higher

Comment (bmwill) [reply .github/workflows/rust.yml]: given we have self-hosted runners big enough now we can just not explicitly set a value for j

**Review by mystenmark — APPROVED** [review]
may need a rebase / re-run of checks to get tests to pass

**Pull request merged**